components/server/ee/src/user/stripe-service.ts

```diff
@@ -93,14 +88,33 @@ export class StripeService {
                 teamId: team.id,
             },
         });
+        // Wait for the customer to show up in Stripe search results before proceeding
+        let attempts = 0;
+        while (!(await this.findCustomerByTeamId(team.id))) {
+            await new Promise((resolve) => setTimeout(resolve, POLL_CREATED_CUSTOMER_INTERVAL_MS));
+            if (++attempts > POLL_CREATED_CUSTOMER_MAX_ATTEMPTS) {
+                throw new Error(`Could not confirm Stripe customer creation for team '${team.id}'`);
+            }
+        }
+        return customer;
+    }
+
+    async setPreferredCurrencyForCustomer(customer: Stripe.Customer, currency: string): Promise<void> {
```

Comment:
How do we ensure that `currency` is a valid value? Does Stripe reject an invalid type of currency?

Reply:
`currency` can be any string. Then, when we choose a Gitpod Product Price in Stripe, we see if one is available in our configuration for that specific string. If there is, great. If not, we pick the `USD` Price.